src/librvnpal/src/posix/writefileheader.c

```diff
@@ -10,65 +10,72 @@
 #include "rvn.h"
 #include "status_codes.h"
 
-
 int32_t
-rvn_write_header (const char *path, void *header, int32_t size,
-		  uint32_t * detailed_error_code) {
-  int32_t rc;
-  bool syncIsNeeded = false;
-  int32_t fd = open (path, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR);
-
-  if (fd == -1) {
-    rc = FAIL_OPEN_FILE;
-    goto error_cleanup;
-  }
-
-  int32_t remaining = size;
-
-  int64_t sz = lseek (fd, 0L, SEEK_END);
-  if (sz == -1) {
-    rc = FAIL_SEEK_FILE;
-    goto error_cleanup;
-  }
-
-  if (lseek (fd, 0L, SEEK_SET) == -1) {
-    rc = FAIL_SEEK_FILE;
-    goto error_cleanup;
-  }
-
-  if (sz != remaining)
-    syncIsNeeded = true;
-
-  while (remaining > 0) {
-      uint64_t written = write (fd, header, (uint64_t) remaining);
-      if (written == -1) {
-    	  rc = FAIL_WRITE_FILE;
-    	  goto error_cleanup;
-    	}
-
-      remaining -= (int) written;
-      header += written;
-  }
-  if (flush_file (fd) == -1) {
-    rc = FAIL_FLUSH_FILE;
-    goto error_cleanup;
-  }
-
-  close (fd);
-  fd = -1;
-
-  if (syncIsNeeded == true)
-    return sync_directory_for (path, detailed_error_code);
-  return SUCCESS;
+rvn_write_header(const char *path,
+                 void *header, int32_t size,
+                 int32_t *detailed_error_code)
+{
+    int32_t rc;
+    bool syncIsNeeded = false;
+    int32_t fd = open(path, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR);
+
+    if (fd == -1)
+    {
+        rc = FAIL_OPEN_FILE;
+        goto error_cleanup;
+    }
+
+    int32_t remaining = size;
+
+    /* do not call rvn_get_file_size to avoid 3 lseek calls, and use only 2 calls */
+    int64_t sz = lseek(fd, 0L, SEEK_END);
+    if (sz == -1)
+    {
+        rc = FAIL_SEEK_FILE;
+        goto error_cleanup;
+    }
+
+    if (lseek(fd, 0L, SEEK_SET) == -1)
+    {
+        rc = FAIL_SEEK_FILE;
+        goto error_cleanup;
+    }
+
+    if (sz != remaining)
+        syncIsNeeded = true;
+
+    while (remaining > 0)
+    {
+        uint64_t written = write(fd, header, (uint64_t)remaining);
```

Comment:
Can just use `pwrite` here and avoid setting calling `lseek` again.
Although, can call `stat` here, which likely will be fastest option